### ISSUE TITLE
Win Key shadowing <Win> + <G/L>   

### ISSUE THREAD

**Issue by gerhard-h** [kmonad/kmonad#629]
1. bug variant `<Win> + <L>  `
this bug is triggert if: 
you map the L key to a different (ordinary) key for exsample <right> on a layer
when you now enter your layer and type `<Win> + <right>` you get `<Win> + <L>` (lock screen) instead
- this ONLY occurs if you use the Win key from your base layer

2. bug variant `<Win> + <G>  `
On the number layer the g key is mapped to  <win> on hold (and any number on tap)
but instead of just holding <win>
`<Win> + <G>` is triggered also (opens the XBox game bar)
- this ONLY  occurs if the G key on the base layer is a dual function key like in the exsample config below
- if you try to get the 1. bug variant also ... you won't  (`<hold Win from base layer> + <tap g on numlayer> `everthing works fine: `<Win> + <0>` )

here a config to reproduce the  G bug
```
(defcfg
     input  (low-level-hook)
     output (send-event-sink 500 30)
     fallthrough true
     allow-cmd false
)

(defsrc
  g 
  spc
)

(defalias
  num  (layer-toggle numbers)
  lnu (tap-hold-next-release 500 spc @num) 
  wn0 (tap-hold-next-release 340 0 rmet)
  geq (tap-hold 200 g S-0)
)
(deflayer qwerty
  @geq
  @lnu
)
(deflayer numbers
  @wn0 
  _
)
```
here a config to reproduce the L bug
```
(defcfg
     input  (low-level-hook)
     output (send-event-sink 500 30)
     fallthrough true
     allow-cmd false
)

(defsrc
  w
  l
  lmet
  spc
)

(defalias
  num  (layer-toggle numbers)
  lnu (tap-hold-next-release 500 spc @num)
)
(deflayer qwerty
  _
  _
  _
  @lnu
)
(deflayer numbers
  lmet
  right 
  _
  _
)
```

**Comment by slotThe** [comment]
Are you sure this is not a faulty keyboard?  Can you run kmonad with `-ldebug` and paste the output (and the keys you presse)?

**Comment by gerhard-h** [comment]
the keyboard is not faulty in any way 

I hold spc to enter number layer, then hold g wich is mapped to 0 on tap and rmet on hold

the log is lying in two ways
1. rmet will not get released
2. g will be pressed

a milder form of this bug exists also in kanata (rmet not stuck and g only fired when no other key is pressed)
I suppose the WIN+G shortcut is caught be windows on a lower level than the keysink

```
Opening KeySink
Initializing Windows key sink
Opening KeySource
Registering low-level Windows keyboard hook
Launching process: emitter_proc


--------------------------------------------------------------------------------
Received event: Press <spc>
Running hooks
Block level set to: 1
Registering untimed hook: 12
Registering 500ms hook: 11

--------------------------------------------------------------------------------
Received event: Press <g>
Running hooks
Registering untimed hook: 14
Registering 126ms hook: 13
Storing event, current store:
 - Press <g>

Cancelling hook: 13
Pushed layer to stack: numbers
 1. numbers
Base-layer: qwerty

Registering untimed hook: 15
Unblocking input stream, rerunning:
Press <g>

Tried cancelling expired hook: 11

--------------------------------------------------------------------------------
Rerunning event: Press <g>
Running hooks
Registering untimed hook: 16
Registering untimed hook: 17
Block level set to: 1
Registering untimed hook: 19
Registering 340ms hook: 18
Cancelling hook: 18
Registering untimed hook: 20
Emitting: Press <rmet>
Unblocking input stream, no stored events

--------------------------------------------------------------------------------
Received event: Press <g>
Running hooks
Registering untimed hook: 21
Registering untimed hook: 22
Registering untimed hook: 23
Registering untimed hook: 24

--------------------------------------------------------------------------------
Received event: Press <g>
Running hooks
Registering untimed hook: 25
Registering untimed hook: 26
Registering untimed hook: 27
Registering untimed hook: 28

--------------------------------------------------------------------------------
Received event: Press <g>
Running hooks
Registering untimed hook: 29
Registering untimed hook: 30
Registering untimed hook: 31
Registering untimed hook: 32

--------------------------------------------------------------------------------
Received event: Press <g>
Running hooks
Registering untimed hook: 33
Registering untimed hook: 34
Registering untimed hook: 35
Registering untimed hook: 36

--------------------------------------------------------------------------------
Received event: Press <g>
Running hooks
Registering untimed hook: 37
Registering untimed hook: 38
Registering untimed hook: 39
Registering untimed hook: 40

--------------------------------------------------------------------------------
Received event: Press <g>
Running hooks
Registering untimed hook: 41
Registering untimed hook: 42
Registering untimed hook: 43
Registering untimed hook: 44

--------------------------------------------------------------------------------
Received event: Press <g>
Running hooks
Registering untimed hook: 45
Registering untimed hook: 46
Registering untimed hook: 47
Registering untimed hook: 48

--------------------------------------------------------------------------------
Received event: Press <g>
Running hooks
Registering untimed hook: 49
Registering untimed hook: 50
Registering untimed hook: 51
Registering untimed hook: 52

--------------------------------------------------------------------------------
Received event: Press <g>
Running hooks
Registering untimed hook: 53
Registering untimed hook: 54
Registering untimed hook: 55
Registering untimed hook: 56

--------------------------------------------------------------------------------
Received event: Press <g>
Running hooks
Registering untimed hook: 57
Registering untimed hook: 58
Registering untimed hook: 59
Registering untimed hook: 60

--------------------------------------------------------------------------------
Received event: Press <g>
Running hooks
Registering untimed hook: 61
Registering untimed hook: 62
Registering untimed hook: 63
Registering untimed hook: 64

--------------------------------------------------------------------------------
Received event: Press <g>
Running hooks
Registering untimed hook: 65
Registering untimed hook: 66
Registering untimed hook: 67
Registering untimed hook: 68

--------------------------------------------------------------------------------
Received event: Press <g>
Running hooks
Registering untimed hook: 69
Registering untimed hook: 70
Registering untimed hook: 71
Registering untimed hook: 72

--------------------------------------------------------------------------------
Received event: Press <g>
Running hooks
Registering untimed hook: 73
Registering untimed hook: 74
Registering untimed hook: 75
Registering untimed hook: 76

--------------------------------------------------------------------------------
Received event: Press <g>
Running hooks
Registering untimed hook: 77
Registering untimed hook: 78
Registering untimed hook: 79
Registering untimed hook: 80

--------------------------------------------------------------------------------
Received event: Press <g>
Running hooks
Registering untimed hook: 81
Registering untimed hook: 82
Registering untimed hook: 83
Registering untimed hook: 84

--------------------------------------------------------------------------------
Received event: Press <g>
Running hooks
Registering untimed hook: 85
Registering untimed hook: 86
Registering untimed hook: 87
Registering untimed hook: 88

--------------------------------------------------------------------------------
Received event: Press <g>
Running hooks
Registering untimed hook: 89
Registering untimed hook: 90
Registering untimed hook: 91
Registering untimed hook: 92

--------------------------------------------------------------------------------
Received event: Press <g>
Running hooks
Registering untimed hook: 93
Registering untimed hook: 94
Registering untimed hook: 95
Registering untimed hook: 96

--------------------------------------------------------------------------------
Received event: Press <g>
Running hooks
Registering untimed hook: 97
Registering untimed hook: 98
Registering untimed hook: 99
Registering untimed hook: 100

--------------------------------------------------------------------------------
Received event: Press <g>
Running hooks
Registering untimed hook: 101
Registering untimed hook: 102
Registering untimed hook: 103
Registering untimed hook: 104

--------------------------------------------------------------------------------
Received event: Press <g>
Running hooks
Registering untimed hook: 105
Registering untimed hook: 106
Registering untimed hook: 107
Registering untimed hook: 108

--------------------------------------------------------------------------------
Received event: Press <g>
Running hooks
Registering untimed hook: 109
Registering untimed hook: 110
Registering untimed hook: 111
Registering untimed hook: 112

--------------------------------------------------------------------------------
Received event: Press <g>
Running hooks
Registering untimed hook: 113
Registering untimed hook: 114
Registering untimed hook: 115
Registering untimed hook: 116

--------------------------------------------------------------------------------
Received event: Press <g>
Running hooks
Registering untimed hook: 117
Registering untimed hook: 118
Registering untimed hook: 119
Registering untimed hook: 120

--------------------------------------------------------------------------------
Received event: Press <g>
Running hooks
Registering untimed hook: 121
Registering untimed hook: 122
Registering untimed hook: 123
Registering untimed hook: 124

--------------------------------------------------------------------------------
Received event: Release <g>
Running hooks
Registering untimed hook: 125
Registering untimed hook: 126
Emitting: Release <rmet>

--------------------------------------------------------------------------------
Received event: Release <spc>
Running hooks
Popped layer from stack: numbers
Base-layer: qwerty

```
